app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -135,19 +138,36 @@ class MessagesController @Inject() (
     }
   }
 
+  private def extractAuditMessageType(messageType: MessageType): EitherT[Future, MessageTypeExtractionError, AuditType] = {
+    val auditType: EitherT[Future, MessageTypeExtractionError, AuditType] =
+      EitherT.fromOption[Future](messageType.auditType, MessageTypeExtractionError.InvalidMessageType(s"$messageType is not a ResponseMessageType"))
+    auditType
+  }
+
   def incomingViaEIS(ids: ConversationId): Action[Source[ByteString, _]] =
     authenticateEISToken.stream(transformer = eisMessageTransformers.unwrap) {
-      implicit request => _ =>
+      implicit request => size =>
         import MessagesController.PresentationEitherTHelper
 
         val (movementId, triggerId) = ids.toMovementAndMessageId
-
         (for {
           messageType <- messageTypeExtractor.extract(request.headers, request.body).asPresentationWithMessageType(None)
+          auditMsg    <- extractAuditMessageType(messageType).asPresentationWithMessageType(None)
```

Comment:
I don't think so we need to send auditType with ```None```

Reply:
That's not what this doing -- it's just for error handling.

That said, this should be `asPresentationWithMessageType(Some(messageType))`.

Reply:
It's not sending the audit type.

I'll change to directly return the presentation error.